### PR TITLE
Support Splunk as Datasource

### LINKED ISSUE #10: Add splunk as a datasource
## 🚀 Feature Proposal

A lot of user reviews may go into splunk. Splunk exposes an API which can directly be used to achieve this.

## Motivation

To track those user reviews which go into splunk.

## Example

Splunk can now become a data source.

### PR DESCRIPTION
**Pull request by kokikathir**
PR checklist:

Resolves #10
- [#10  ] Github Issue associated with branch.
- [ ] Code passes linting through autopep8.
- [ Yes ] PR branch is up-to-date with latest master branch.

### CODE REVIEW

**Comment by kokikathir** [comment]
@MadaraUchiha-314,  Created this PR to query data from splunk. I have used splunk SDK for python as it had in built capabilities for querying. Kindly review

File: fawkes/configs/app_config.py [hunk 1]
```diff
@@ -266,6 +267,29 @@ def __init__(self, config):
...
+
+    """
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.host = config["host"]
+        self.port = config["port"]
+        self.username = config["username"]
+        self.password = config["password"]
+        self.query = config["query"]
```
Comment (MadaraUchiha-314) [inline fawkes/configs/app_config.py:287]: Kindly add the extra attributes to the docstring.

File: fawkes/configs/app_config.py [hunk 2]
```diff
@@ -266,6 +267,29 @@ def __init__(self, config):
...
+        self.earliest = config["earliest_time"]
+        self.latest = config['latest_time']
```
Comment (MadaraUchiha-314) [inline fawkes/configs/app_config.py:289]: Can't this be a part if the query itself ?

File: fawkes/fetch/fetch.py [hunk 3]
```diff
@@ -62,6 +63,10 @@ def fetch_reviews(fawkes_config_file = constants.FAWKES_CONFIG_FILE):
                     reviews = appstore.fetch(
                         review_channel
                     )
+                elif review_channel.channel_type == ReviewChannelTypes.SPLUNK:
```
Comment (MadaraUchiha-314) [inline fawkes/fetch/fetch.py:66]: Can you add an example of splunk config in the https://github.com/intuit/fawkes/blob/master/app/sample-mint-config.json

You can keep the `is_channel_enabled` as false, for it to not break the unit tests.

Comment (MadaraUchiha-314) [reply fawkes/fetch/fetch.py:66]: Pls feel free to say no, if it doesn't make sense.

File: fawkes/fetch/splunk.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        query = 'search {} earliest=-{} latest={}'.format(review_channel.query, review_channel.earliest, review_channel.latest)
+
+        stream = service.jobs.export(query)
+
+        reader = results.ResultsReader(stream)
+        for result in reader:
+            if isinstance(result, dict):
+                review = json.loads(result["_raw"])
+                reviews.append(review)
+            elif isinstance(result, results.Message):
```
Comment (MadaraUchiha-314) [inline fawkes/fetch/splunk.py:26]: What exactly is this scenario ? Why are we just printing the message in this case ?

File: fawkes/configs/app_config.py (hunk 1 above)
Comment (kokikathir) [reply fawkes/configs/app_config.py:304]: This is done

File: fawkes/configs/app_config.py (hunk 2 above)
Comment (kokikathir) [reply fawkes/configs/app_config.py]: Updated the query to contain earliest and latest. also referenced the Splunk documentation for the same

File: fawkes/fetch/splunk.py (hunk 4 above)
Comment (kokikathir) [reply fawkes/fetch/splunk.py]: We dont need this else block. removed it

File: fawkes/fetch/fetch.py (hunk 3 above)
Comment (kokikathir) [reply fawkes/fetch/fetch.py:66]: Added the splunk channel in sample config, with channel_enabled=false

**Comment by kokikathir** [comment]
@MadaraUchiha-314 Implemented the review comments. Please review

**Review by MadaraUchiha-314 — APPROVED** [review]
LGTM. Thanks a lot for the contribution @kokikathir

**Pull request merged**